src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -47,11 +47,11 @@ class TaskKiller @Inject() (
             await(killService.killInstances(activeInstances, KillReason.KillingTasksViaApi)): @silent
           } else {
             if (activeInstances.nonEmpty) {
-              val setGoalFutures = foundInstances.map(i => instanceTracker.setGoal(i.instanceId, if (runSpec.isResident) Goal.Stopped else Goal.Decommissioned))
-              await(Future.sequence(setGoalFutures))
+              // This is legit. We don't adjust the goal, since that should stay whatever it is.
+              // However we kill the tasks associated with these instances directly via the killService.
```

Comment:
Why is that in this pull request? It was so nice a clean when they were separated!

Reply:
Anyways, how is the goal set?

Reply:
yes

Reply:
Yes; this code belongs here. There's an integration test checking that functionality.

Reply:
Thanks for clarifying.